web/server/codechecker_server/server.py

```diff
@@ -1100,10 +1102,39 @@ def start_server(config_directory, package_data, port, config_sql_server,
                      "created at '%s'", server_cfg_file)
             shutil.copyfile(example_cfg_file, server_cfg_file)
 
+    def check_callback_url_format(provider_name: str, callback_url: str):
+        """
+        check the format of callback url using regex
+        """
+        if "@" in provider_name:
+            LOG.warning(f"provider {provider_name} contains '@' "
+                        "which is not allowed, turning off provider.")
+            return None
+        protocol = "http(s|)"
+        website = "[a-zA-Z0-9.-_]+([:][0-9]{2,5}|)"
+        paths = "login[/]OAuthLogin"
+
+        pattern_str = f"^{protocol}://{website}/{paths}/{provider_name}$"
+        pattern = re.compile(pattern_str)
+        match = pattern.match(callback_url)
+        if match is None:
+            LOG.warning("Configuration format of callback_url is "
+                        f"invalid for provider {provider_name}."
```

Comment:
Add an extra space to the end so the next sentence is separate.
Also, this message should be printed only if the given provider is enabled.